### ISSUE TITLE
Improve example player move so when pressing against walls the camera doesn't shake

### ISSUE THREAD

**Issue by subsoap** [astrochili/defold-trenchbroom#7]
This may give bad impression to people considering this tool if they don't know how to fix it themselves

When pressing against walls and not moving camera shakes, this shouldn't happen to look better

**Comment by astrochili** [comment]
It's problem of Kinematic Walker I think but also it's possible to improve the collision geometry. I will check it.

**Comment by astrochili** [comment]
https://user-images.githubusercontent.com/4752473/207128926-24c740d0-051a-41ac-b441-f3f76fcbe8bc.mp4

**Comment by subsoap** [comment]
@astrochili 

What I see with current version of project/Defold on Windows 11. Same behavior on HTML5 builds.


https://user-images.githubusercontent.com/409170/207750563-2ebfc036-3508-4e9c-9359-b3efe0c4d557.mp4

**Comment by subsoap** [comment]
> Interesting! Is it the same with the [public demo](https://astronachos.com/defold/trenchbroom/)?

Indeed! Same behavior on Windows 11 with Chrome and Firefox. I'll test on a Mac...

**Comment by subsoap** [comment]
So! I just tested on a macOS device on Chrome and it's like in your video! I wonder why. @britzl any idea?

**Comment by subsoap** [comment]
macOS / Linux have same behavior that you showed. Different (bad) on Windows somehow for both .exe and HTML5.

**Comment by britzl** [comment]
> So! I just tested on a macOS device on Chrome and it's like in your video! I wonder why. @britzl any idea?

Not sure. Shader precision?

**Comment by astrochili** [comment]
It is very likely related to kinematic physics detection and frame update, so the cause is [kinematic walker](https://github.com/astrochili/defold-kinematic-walker) library.

And maybe not Windows related, but performance related? I have only one mac, unfortunately. Maybe the same bug can be reproduced [here](https://astronachos.com/defold/kinematic-walker/) too, running on Windows?

**Comment by subsoap** [comment]
I asked on Discord it seems people with more powerful GPUs see the stuttering people with older GPUs do not. Could be related to the recent vsync changes/KW maybe not updating to the new function.

**Comment by subsoap** [comment]
> Maybe the same bug can be reproduced [here](https://astronachos.com/defold/kinematic-walker/) too

Yes!


https://user-images.githubusercontent.com/409170/208310476-2bc5ed5c-f075-46ea-b836-eece64326669.mp4
